src/nnet3/nnet-simple-component.cc

```diff
@@ -128,6 +130,11 @@ void* DropoutComponent::Propagate(const ComponentPrecomputedIndexes *indexes,
 
   BaseFloat dropout = dropout_proportion_;
   KALDI_ASSERT(dropout >= 0.0 && dropout <= 1.0);
+  if (test_mode_) {
+    out->Set(1.0 - dropout);
```

Comment:
This would not work correctly for in-place propagation, if we ever change the code to make that possible, so I think it would be nicer to replace this code with:
```
  out->CopyFromMat(in);
  out->Scale(1.0  - dropout);
  return NULL;
```